my-app/src/components/Transactions/Transactions.tsx

```diff
@@ -1,5 +1,165 @@
+import { useSelector } from "react-redux";
+import { RootState } from "../../store/store";
+import { selectUniqueCategoriesTransactions } from "../../utils/transactionsUtils.ts";
+import { SelectSortBy } from "../SelectSortBy/SelectSortBy.tsx";
+import { InputSearch } from "../InputSearch/InputSearch.tsx";
 
-export const Transactions = () => {
-  return <></>;
+interface TransactionsProps {
+  avatar: string;
+  name: string;
+  category: string;
+  date: string;
+  amount: number;
+  recurring: boolean;
+}
+
+const sortFunctions = {
+  All: (transactions: TransactionsProps[]) => transactions,
+  Highest: (transactions: TransactionsProps[]) =>
+    [...transactions].sort((a, b) => b.amount - a.amount),
+  Lowest: (transactions: TransactionsProps[]) =>
+    [...transactions].sort((a, b) => a.amount - b.amount),
+  Oldest: (transactions: TransactionsProps[]) =>
+    [...transactions].sort(
+      (a, b) => new Date(a.date).getTime() - new Date(b.date).getTime()
+    ),
+  Latest: (transactions: TransactionsProps[]) =>
+    [...transactions].sort(
+      (a, b) => new Date(b.date).getTime() - new Date(a.date).getTime()
+    ),
+  "A to Z": (transactions: TransactionsProps[]) =>
+    [...transactions].sort((a, b) => a.name.localeCompare(b.name)),
+  "Z to A": (transactions: TransactionsProps[]) =>
+    [...transactions].sort((a, b) => b.name.localeCompare(a.name)),
 };
+
+export const Transactions: FC = () => {
+  const { transactions } = useSelector((state: RootState) => state.data);
+  const categoriesTransactions = useSelector((state: RootState) =>
+    selectUniqueCategoriesTransactions(state.data)
+  );
+
+  const [page, setPage] = useState<number>(0);
+  const [rowsPerPage] = useState<number>(10);
+  const [searchTerm, setSearchTerm] = useState<string>("");
+  const [selectedSort, setSelectedSort] = useState<string>("All");
+  const [selectedCategory, setSelectedCategory] = useState<string>("All");
+
+  const tableHeaders = ["Recipient", "Category", "Transaction Date", "Amount"];
+
+  const filteredByCategory =
+    selectedCategory === "All"
+      ? transactions
+      : transactions.filter(
+          (transaction) => transaction.category === selectedCategory
+        );
+
+  const filteredBySearch = filteredByCategory.filter((transaction) =>
+    transaction.name.toLowerCase().includes(searchTerm.toLowerCase())
+  );
+
+  const sortedTransactions = sortFunctions[selectedSort](filteredBySearch);
+
+  const handleChangePage = (event: unknown, newPage: number) => {
+    setPage(newPage);
+  };
+
+  return (
+    <StyledWrapper>
+      <SectionHeader>
+        <h1>Transactions</h1>
+      </SectionHeader>
+      <StyledWrapperDetails>
+        <StyledWrapperDetailsHeader>
+          <div>
+            <InputSearch
+              searchTerm={searchTerm}
+              onChange={(value) => setSearchTerm(value)}
+            />
+          </div>
+          <div>
+            <SelectSortBy
+              label={"Category"}
```

Comment:
```suggestion
              label="Category"
```

🤷  ;-)

Let's check your `prettier` settings during the lesson :-)